src/Castle.Windsor/Core/Internal/EnumerableExtensions.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright 2004-2011 Castle Project - http://www.castleproject.org/
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.Core.Internal
+{
+	using System;
+	using System.Collections.Generic;
+
+	static class EnumerableExtensions
+	{
+		public static Y[] ConvertAll<X, Y>(this X[] items, Func<X, Y> converter)
+		{
+			var results = new List<Y>();
+
+			foreach (var item in items)
+			{
+				var y = converter(item);
+
+				results.Add(y);
+			}
+
+			return results.ToArray();
```

Comment:
Any idea how often this method can get called? Would be much nicer on the GC to allocate an array and populate it.